src/overlays/actors/ovl_Oceff_Wipe4/z_oceff_wipe4.c

```diff
@@ -28,12 +27,87 @@ const ActorInit Oceff_Wipe4_InitVars = {
     (ActorFunc)OceffWipe4_Draw,
 };
 
-#endif
+#include "assets/overlays/ovl_Oceff_Wipe4/ovl_Oceff_Wipe4.c"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe4/OceffWipe4_Init.s")
+s32 D_8099E780;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe4/OceffWipe4_Destroy.s")
+void OceffWipe4_Init(Actor* thisx, PlayState* play) {
+    OceffWipe4* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe4/OceffWipe4_Update.s")
+    Actor_SetScale(&this->actor, 0.1f);
+    this->counter = 0;
+    this->actor.world.pos = GET_ACTIVE_CAM(play)->eye;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe4/OceffWipe4_Draw.s")
+void OceffWipe4_Destroy(Actor* thisx, PlayState* play) {
+    OceffWipe4* this = THIS;
+
+    func_80115D5C(&play->state);
+    play->msgCtx.unk120B0 = 0;
+}
+
+void OceffWipe4_Update(Actor* thisx, PlayState* play) {
+    OceffWipe4* this = THIS;
+
+    this->actor.world.pos = GET_ACTIVE_CAM(play)->eye;
+    if (this->counter < 50) {
+        this->counter++;
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void OceffWipe4_Draw(Actor* thisx, PlayState* play) {
+    u32 scroll = play->state.frames & 0xFFF;
+    OceffWipe4* this = THIS;
+    f32 z;
+    u8 alpha;
+    s32 pad[2];
+    Vec3f eye = GET_ACTIVE_CAM(play)->eye;
+    Vtx* vtxPtr;
+    Vec3f quakeOffset;
+
+    Camera_GetQuakeOffset(&quakeOffset, GET_ACTIVE_CAM(play));
+
+    if (this->counter < 16) {
+        z = Math_SinS(this->counter * 1024) * 1220.0f;
```

Comment:
```suggestion
        z = Math_SinS(this->counter * 0x400) * 1220.0f;
```
Since this is an angle